src/UI/templates/js/Core/ui.js

```diff
@@ -0,0 +1,51 @@
+var il = il || {};
+il.UI = il.UI || {};
+
+(function($, UI) {
+
+	UI.core = (function ($) {
+
+		/**
+		 * Replace a component or parts of a component using ajax call
+		 *
+		 * @param id component id
+		 * @param url replacement url
+		 * @param marker replacement marker ("component", "content", "header", ...)
+		 */
+		var replaceContent = function (id, url, marker) {
```

Comment:
Maybe use in the public interface the functions content, component, ... to make the options very explizit and use an internal replace function taking marker as argument. So you would have on the interface something like UI.core.replace.content(myId, myUrl), which internally calls performReplace(myId, myUrl, "content").

So replaceContent would be split into the functions content, header, component .... which are all part of the "replace" scope. Internally this scope would also have some internalReplace function which would perform most of the action.